### PR TITLE
Allow separate config files for parameters with class type when LightningCLI is in subclass_mode=False

### PR DESCRIPTION
**Pull request by mauvilsa**
## What does this PR do?

This enables the possibility of having separate config files for parameters whose type hint is a class when the `LightningCLI` is in `subclass_mode=False`. For `subclass_mode=True` this option is enabled by default so this change also makes it more consistent.

For example for a CLI like:

```python
from torchaudio.transforms import MelSpectrogram
from pytorch_lightning.utilities.cli import LightningCLI
import pytorch_lightning as pl

class PrxMelDataModule(pl.LightningDataModule):
    def __init__(self, mel_spec: MelSpectrogram):
        ...

class HifiGAN(pl.LightningModule):
    def __init__(self, mel_spec: MelSpectrogram):
        ...

if __name__ == "__main__":
    LightningCLI(
        model_class=HifiGAN,
        datamodule_class=PrxMelDataModule,
    )
```

it would be possible to have a config like:

```yaml
data:
  mel_spec: mel_spec_config.yml
model:
  mel_spec: mel_spec_config.yml
```

Fixes https://github.com/omni-us/jsonargparse/issues/95#issuecomment-944367550

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by tchaton** [comment]
Dear @mauvilsa,

We are releasing v1.5 by tomorrow. Do you believe you could get this PR by then ?

Best,
T.C

**Comment by mauvilsa** [comment]
> We are releasing v1.5 by tomorrow. Do you believe you could get this PR by then ?

Yes, sure. This is a small change.

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/utilities/cli.py [hunk 1]
```diff
@@ -155,7 +155,11 @@ def add_lightning_class_args(
             if subclass_mode:
                 return self.add_subclass_arguments(lightning_class, nested_key, fail_untyped=False, required=required)
             return self.add_class_arguments(
-                lightning_class, nested_key, fail_untyped=False, instantiate=not issubclass(lightning_class, Trainer)
+                lightning_class,
+                nested_key,
+                fail_untyped=False,
+                instantiate=not issubclass(lightning_class, Trainer),
+                sub_configs=True,
```
Comment (carmocca) [inline pytorch_lightning/utilities/cli.py:162]: Does this need to bump the min jsonargparse version?

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/utilities/cli.py (hunk 1 above)
Comment (mauvilsa) [reply pytorch_lightning/utilities/cli.py:162]: No.

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Pull request merged**